tests/TestCase/Controller/ComponentRegistryTest.php

```diff
@@ -209,6 +209,36 @@ public function testUnload()
     }
 
     /**
+     * Test that unloading a none existing component triggers an error.
+     *
+     * This should produce an "Object "Foo" was not loaded before." error
+     * which gets thrown as a \PHPUnit\Framework\Error\Error Exception by PHPUnit.
+     *
+     * @expectedException \PHPUnit\Framework\Error\Error
+     * @expectedExceptionMessage Object "Foo" was not loaded before.
+     * @return void
+     */
+    public function testUnloadUnknown()
+    {
+        $this->Components->unload('Foo');
```

Comment:
Can't you just test `ObjectRegistry` itself? If a class does not overwrite a method, it does not need a test IMO.